### PR TITLE
Move common Makefile targets to Makefile.common

### PR DESCRIPTION
**Pull request by rafaelvanoni**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,70 +1,8 @@
...
-ARCH ?= $(BUILDARCH)
-
-# canonicalized names for target architecture
-ifeq ($(ARCH),aarch64)
-	override ARCH=arm64
-endif
-ifeq ($(ARCH),x86_64)
-	override ARCH=amd64
-endif
+get_common:=$(shell wget -nv https://raw.githubusercontent.com/rafaelvanoni/common/master/Makefile.common -O Makefile.common)
```
Comment (fasaxc) [inline Makefile]: Presumably a temporary hack?

File: pkg/syncclient/sync_client.go [hunk 2]
```diff
@@ -209,6 +209,7 @@ func (s *SyncerClient) connect(cxt context.Context) error {
 			return err
 		}
 		tlsConfig := tls.Config{Certificates: []tls.Certificate{cert}}
+		tlsConfig.MaxVersion = tls.VersionTLS12
```
Comment (fasaxc) [inline pkg/syncclient/sync_client.go]: What's this doing here?  Needs a comment to explain it if nothing else

File: Makefile (hunk 1 above)
Comment (rafaelvanoni) [reply Makefile]: Of course

File: pkg/syncclient/sync_client.go (hunk 2 above)
Comment (rafaelvanoni) [reply pkg/syncclient/sync_client.go]: go 1.13 defaults to TLS13, which causes some UT failures

Comment (tmjd) [reply pkg/syncclient/sync_client.go]: Please create a Core story to revisit this and allow/enable TLS 1.3.

File: Makefile [hunk 3]
```diff
@@ -239,11 +70,23 @@ build-all: $(addprefix sub-build-,$(VALIDARCHES))
 sub-build-%:
 	$(MAKE) build ARCH=$*
 
+# Build mounts for running in "local build" mode. This allows an easy build using local development code,
+# assuming that there is a local checkout of libcalico in the same directory as this repo.
+PHONY:local_build
```
Comment (tmjd) [inline Makefile]: Should this be inside the `ifdef` below?

Comment (rafaelvanoni) [reply Makefile]: We need the `local_build` rule for when we _don't_ run a local build, so the PHONY label needs to be there.

Comment (fasaxc) [reply Makefile]: Compare with the [change I made to the felix build](https://github.com/projectcalico/felix/commit/27a087ca019db79a12849aa06843c5b71f317c2b#diff-b67911656ef5d18c4ae36cb6741b7965) to change the local_build dependency into a variable.  

Depending on a phony is a big smell in a makefile, it always forces a rebuild of the target.  Removing that took the FV cycle time down from 50s to 20s, which is a big win.

Comment (rafaelvanoni) [reply Makefile]: FIxed

**Pull request merged**